tests/integration-all/api-gateway-websocket/tests.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const path = require('path');
+const AWS = require('aws-sdk');
+const _ = require('lodash');
+const { expect } = require('chai');
+
+const { getTmpDirPath, readYamlFile, writeYamlFile } = require('../../utils/fs');
+const {
+  region,
+  // confirmCloudWatchLogs,
+  createTestService,
+  deployService,
+  removeService,
+} = require('../../utils/misc');
+const {
+  createApi,
+  deleteApi,
+  getRoutes,
+  createStage,
+  deleteStage,
+} = require('../../utils/api-gateway-v2');
+
+const CF = new AWS.CloudFormation({ region });
+
+describe('AWS - API Gateway Websocket Integration Test', function() {
+  this.timeout(1000 * 60 * 10); // Involves time-taking deploys
+  let serviceName;
+  let stackName;
+  let tmpDirPath;
+  let serverlessFilePath;
+  let websocketApiId;
+  const stage = 'dev';
+
+  before(() => {
+    tmpDirPath = getTmpDirPath();
+    console.info(`Temporary path: ${tmpDirPath}`);
+    serverlessFilePath = path.join(tmpDirPath, 'serverless.yml');
+    const serverlessConfig = createTestService(tmpDirPath, {
+      templateDir: path.join(__dirname, 'service'),
+    });
+    serviceName = serverlessConfig.service;
+    stackName = `${serviceName}-${stage}`;
+    console.info(`Deploying "${stackName}" service...`);
+    deployService(tmpDirPath);
+    // create an external websocket API
+    const externalWebsocketApiName = `${stage}-${serviceName}-ext-api`;
+    return createApi(externalWebsocketApiName).then(wsApiMeta => {
```

Comment:
Creation of external API, should be moved into `before` of `when using an existing websocket API` test

I know it was copied from REST API test, where it's like that, but it's a test setup issue which I plan to fix at convient moment (e.g. just running `should expose an accessible websocket endpoint` test shouldn't really demand creating such external websockets API, and currently it does, which is not good)

Reply:
done.